apps/src/music/utils/Loader.js

```diff
@@ -0,0 +1,101 @@
+// Utilities for retrieving various types of data for Music Lab.
+
+import {getStore} from '@cdo/apps/redux';
+import AppConfig from '../appConfig';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressRedux';
+
+export const baseUrl = 'https://curriculum.code.org/media/musiclab/';
+
+// Loads a sound library JSON file.
+export const loadLibrary = async () => {
+  if (AppConfig.getValue('local-library') === 'true') {
+    const localLibraryFilename = 'music-library';
+    const localLibrary = require(`@cdo/static/music/${localLibraryFilename}.json`);
+    return localLibrary;
+  } else {
+    const libraryParameter = AppConfig.getValue('library');
+    const libraryFilename = libraryParameter
+      ? `music-library-${libraryParameter}.json`
+      : 'music-library.json';
+    const response = await fetch(baseUrl + libraryFilename);
+    const library = await response.json();
+    return library;
+  }
+};
+
+// Loads a progression file.  This file can be used for prototyping a progression
+// before it's split into proper levels.
+const loadProgressionFile = async () => {
+  if (AppConfig.getValue('local-progression') === 'true') {
+    const defaultProgressionFilename = 'music-progression';
+    const progression = require(`@cdo/static/music/${defaultProgressionFilename}.json`);
+    return progression;
+  } else {
+    const progressionParameter = AppConfig.getValue('progression');
+    const progressionFilename = progressionParameter
+      ? `music-progression-${progressionParameter}.json`
+      : 'music-progression.json';
+    const response = await fetch(baseUrl + progressionFilename);
+    const progression = await response.json();
+    return progression;
+  }
+};
+
+// Loads level data from the dashboard, for a script level or a level.
+const loadLevelData = async levelSource => {
+  const state = getStore().getState();
```

Comment:
I'm a bit wary of accessing the Redux store directly in non-react files (I know we do it elsewhere)...could we instead maybe have a selector or selectors for computing the `levelDataPath`s based on state?

Reply:
Sure thing.  First pass at it is now done.